tests/cli/test_deploy.py

```diff
@@ -4878,6 +4878,38 @@ async def test_work_pool_image_already_exists_not_updated_after_adding_build_doc
         }
 
 
+class TestDeployInfraOverrides:
+    async def test_uses_infra_overrides(self, project_dir, prefect_client):
+        await prefect_client.create_work_pool(
+            WorkPoolCreate(name="test-pool", type="test")
+        )
+        await run_sync_in_worker_thread(
+            invoke_and_assert,
+            command=(
+                "deploy ./flows/hello.py:my_flow -n test-name -p test-pool --version"
+                " 1.0.0 -v env=prod -t foo-bar --variable"
+                ' \'{"resources":{"limits":"cpu"}}\''
+            ),
+            expected_code=0,
+            expected_output_contains=[
+                "An important name/test-name",
+                "prefect worker start --pool 'test-pool'",
+            ],
+        )
+
+        deployment = await prefect_client.read_deployment_by_name(
+            "An important name/test-name"
+        )
+        assert deployment.name == "test-name"
+        assert deployment.work_pool_name == "test-pool"
+        assert deployment.version == "1.0.0"
+        assert deployment.tags == ["foo-bar"]
+        assert deployment.infra_overrides == {
+            "env": "prod",
+            "resources": {"limits": "cpu"},
+        }
+
```

Comment:
Nice! Wondering if it might be a good idea to throw in some test cases for `_extract_variable` specifically for raising a `TypeError` and `ValueError`

Reply:
Yeah, absolutely!